duckdb-vortex/src/expr/expr.cpp

```diff
@@ -0,0 +1,216 @@
+	case ExpressionType::COMPARE_GREATERTHAN:
+		return vortex::expr::Kind_BinaryOp_Gt;
+	case ExpressionType::COMPARE_LESSTHANOREQUALTO:
+		return vortex::expr::Kind_BinaryOp_Lte;
+	case ExpressionType::COMPARE_GREATERTHANOREQUALTO:
+		return vortex::expr::Kind_BinaryOp_Gte;
+	case ExpressionType::CONJUNCTION_AND:
+		return vortex::expr::Kind_BinaryOp_And;
+	case ExpressionType::CONJUNCTION_OR:
+		return vortex::expr::Kind_BinaryOp_Or;
+	default:
+		throw Exception(ExceptionType::NOT_IMPLEMENTED, "impl");
+	}
+}
+
+vortex::dtype::DType *into_vortex_dtype(Arena &arena, const LogicalType &type_, bool nullable) {
+	auto *dtype = Arena::Create<vortex::dtype::DType>(&arena);
+	switch (type_.id()) {
+	case LogicalTypeId::INVALID:
+	case LogicalTypeId::SQLNULL:
+		dtype->mutable_null();
+		return dtype;
+	case LogicalTypeId::BOOLEAN:
+		dtype->mutable_binary()->set_nullable(nullable);
+		return dtype;
+	case LogicalTypeId::TINYINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::I8);
+		return dtype;
+	case LogicalTypeId::SMALLINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::I16);
+		return dtype;
+	case LogicalTypeId::INTEGER:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::I32);
+		return dtype;
+	case LogicalTypeId::BIGINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::I64);
+		return dtype;
+	case LogicalTypeId::UTINYINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::U8);
+		return dtype;
+	case LogicalTypeId::USMALLINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::U16);
+		return dtype;
+	case LogicalTypeId::UINTEGER:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::U32);
+		return dtype;
+	case LogicalTypeId::UBIGINT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::U64);
+		return dtype;
+	case LogicalTypeId::FLOAT:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::F32);
+		return dtype;
+	case LogicalTypeId::DOUBLE:
+		dtype->mutable_primitive()->set_nullable(nullable);
+		dtype->mutable_primitive()->set_type(vortex::dtype::F64);
+		return dtype;
+	case LogicalTypeId::CHAR:
+	case LogicalTypeId::VARCHAR:
+		dtype->mutable_utf8()->set_nullable(nullable);
+		return dtype;
+	case LogicalTypeId::BLOB:
+		dtype->mutable_binary()->set_nullable(nullable);
+		return dtype;
+	default:
+		break;
+	}
+	throw Exception(ExceptionType::NOT_IMPLEMENTED, "into_vortex_dtype", {{"id", type_.ToString()}});
+}
+
+vortex::scalar::Scalar *into_null_scalar(Arena &arena, LogicalType &type_) {
```

Comment:
Preference for going with `logical_type` over var names with `_` as a suffix, even if longer.